### PR TITLE
Correct Reservoir H; Merge IO changes into v01

### PR DESCRIPTION
**Pull request by jameshalgren**
This PR is *mostly* a series of changes merged into v01 that do not directly affect the v02 routing. Eventually (soon), the IO changes  should be ported over to v02. 

More importantly, there was a key bug fix regarding handling of depth in the reservoir fortran file. This will directly affect the results of reservoir simulations in v02, but should not change the compilation process or function signatures.

### CODE REVIEW

File: src/fortran_routing/mc_pylink_v00/MC_singleSeg_singleTS/MCsingleSegStime_f2py_NOLOOP.f90 [hunk 1]
```diff
@@ -173,7 +173,8 @@ subroutine secant_h0(z, bw, bfd, twcc, s0, n, ncc, dt, dx, &
     real(prec), intent(in) :: dt, dx
     real(prec), intent(in) :: qdp, ql, qup, quc
     real(prec), intent(in) :: h_0  !, refQj_0
-    real(prec), intent(out) :: WPC, Qj_0, C1, C2, C3, C4
+    real(prec), intent(out) :: Qj_0, C1, C2, C3, C4
+    real(prec), intent(inout) :: WPC
```
Comment (jameshalgren) [inline src/fortran_routing/mc_pylink_v00/MC_singleSeg_singleTS/MCsingleSegStime_f2py_NOLOOP.f90:177]: I didn't realize these `(inout)` WPC variables were still in here. I am trying to figure out why Juzer feels so strongly about the WPC initialization. Let me check with him again.

File: src/fortran_routing/mc_pylink_v00/Reservoir_singleTS/module_levelpool.f90 [hunk 2]
```diff
@@ -71,27 +73,31 @@ real :: dh, dh1, dh2, dh3      ! Depth in weir, and height function for 3 order
...
         !pool, .....
         if (LAKE_OPT == 1) then     ! If-block for simple pass through scheme....
 
            qo1 = qi1                 ! Set outflow equal to inflow at current time
-           H = Htmp                  ! Set new lake water elevation to incoming lake el.
+           H = H0                    ! Set new lake water elevation to incoming lake el.
 
         else if (LAKE_OPT == 2) then   ! If-block for Chow et al level pool scheme
 
+           H = H0                    ! Set new lake water elevation to incoming lake el.
```
Comment (jameshalgren) [inline src/fortran_routing/mc_pylink_v00/Reservoir_singleTS/module_levelpool.f90:92]: This `H=H0` was the critical change. Without this, the depth never comes into the equation.

File: src/fortran_routing/mc_pylink_v00/Reservoir_singleTS/reservoirs_nwm.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    maxh=None,
+    wc=None,
+    wl=None,
+    dl=None,
+    oe=None,
+    oc=None,
+    oa=None,
+    h0=None,
+):
+    """This test function demonstrates a simple connection between
```
Comment (jameshalgren) [inline src/fortran_routing/mc_pylink_v00/Reservoir_singleTS/reservoirs_nwm.py:57]: I updated this python script to be a very brief demonstration of how the module is used with f2py.

File: src/fortran_routing/mc_pylink_v00/MC_singleSeg_singleTS/MCsingleSegStime_f2py_NOLOOP.f90 (hunk 1 above)
Comment (groutr) [reply src/fortran_routing/mc_pylink_v00/MC_singleSeg_singleTS/MCsingleSegStime_f2py_NOLOOP.f90:177]: This detail has no effect on how we wrap the fortran code.

**Pull request merged**